oozie/oozie.sh

```diff
@@ -25,16 +27,86 @@
 
 set -euxo pipefail
 
+OS_NAME=$(lsb_release -is | tr '[:upper:]' '[:lower:]')
+distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+readonly OS_NAME
+
 # Use Python from /usr/bin instead of /opt/conda.
 export PATH=/usr/bin:$PATH
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+
+# Detect dataproc image version from its various names
+if (! test -v DATAPROC_IMAGE_VERSION) && test -v DATAPROC_VERSION; then
+  DATAPROC_IMAGE_VERSION="${DATAPROC_VERSION}"
+fi
+
+case "${DATAPROC_IMAGE_VERSION}" in
+  "1.3" | "1.4" | "1.5" | "2.0" )
+    curator_version="2.13.0"
+    curator_src="/usr/lib/hadoop/lib"
+    ;;
+  "2.1")
+#    curator_version="5.2.0"
+    curator_version="2.13.0"
+    curator_src="/usr/lib/spark/jars"
+    ;;
+  *)
+    echo "unsupported DATAPROC_IMAGE_VERSION: ${DATAPROC_IMAGE_VERSION}" >&2
+    exit 1
+    ;;
+esac
+
+# Use Python from /usr/bin instead of /opt/conda.
+export PATH=/usr/bin:$PATH
+export DATAPROC_IMAGE=$(/usr/share/google/get_metadata_value image)
```

Comment:
@cjac - how will these additional parameters be set by the user? I don't see them documented in README.md